dbms/src/Server/Server.cpp

```diff
@@ -219,8 +219,13 @@ int Server::main(const std::vector<std::string> & args)
 		{
 			rlim_t old = rlim.rlim_cur;
 			rlim.rlim_cur = rlim.rlim_max;
+			#ifndef __APPLE__
 			if (setrlimit(RLIMIT_NOFILE, &rlim))
 				throw Poco::Exception("Cannot setrlimit");
+			#else
+			LOG_INFO(log, "Setting nofile limit is not supported on MacOS X, please run 'launchctl limit maxfiles " << rlim.rlim_max << "' (current is " << old << ")");
```

Comment:
I'll add proper manual about how to deal with this on MacOS X, because it's a little bit tricky since 10.11.